pkg/config/resource.go

```diff
@@ -43,6 +43,9 @@ func NewResource(version, kind, terraformResourceType string, opts ...ResourceOp
 		Kind:                  kind,
 		TerraformResourceType: terraformResourceType,
 		TerraformIDFieldName:  "id",
+		ExternalName: ExternalName{
+			ConfigureFn: NopConfigureWithName,
```

Comment:
I had a hard time understanding the purpose of this option in the beginning because it sounds like we are configuring external name, like providing a function to build external name. However, it is almost the opposite.
Do you think if we could find a more intuitive name this? Something like `ExternalName.SetTerraformArgsFn` or at least `ExternalName.ConsumeFn`.

Reply:
Hmm how does `SetNameAttributeFn` sound?

Reply:
Yeah sounds better. Considering we need to set fields like [cluster_identifier](https://github.com/crossplane-contrib/provider-tf-aws/blob/5509c10d768622c3631615947cf7b18086a58aa3/config/rds/config.go#L32), I didn't feel great about having _Name_ there. May be `SetAttributesFn` or `SetArgumentsFn` ? The reason why I suggest Argument was, we would always need to set field(s) from [the Arguments section](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/vpc#argument-reference) in Terraform docs.

Reply:
I see. I don't want it as generic as `SetAttributes` though. So I changed it to `SetIdentifierArgumentFn` which is more precise.

Reply:
Sounds great, thanks!